disable -m32 on ppc (gcc 3.3.6 does not like it)

revert 'returns_twice' attribute for grub_setjmp.
http://repo.or.cz/w/grub2.git/commitdiff/2c7825edcb18e9d0680a953f1475ef2caf6b0f0f

--- grub-1.98/configure.ac~	2010-03-06 20:51:37.000000000 +0000
+++ grub-1.98/configure.ac	2010-07-19 13:59:46.468351205 +0000
@@ -114,7 +114,7 @@
 esac
 
 case "$target_cpu" in
-  i386 | powerpc) target_m32=1 ;;
+  i386) target_m32=1 ;;
   x86_64 | sparc64) target_m64=1 ;;
 esac
 
--- grub-1.98/include/grub/powerpc/setjmp.h~	2010-03-06 20:51:37.000000000 +0000
+++ grub-1.98/include/grub/powerpc/setjmp.h	2010-07-19 16:37:03.616853413 +0000
@@ -21,7 +21,7 @@
 
 typedef unsigned long grub_jmp_buf[20];
 
-int grub_setjmp (grub_jmp_buf env) __attribute__ ((returns_twice));
+int grub_setjmp (grub_jmp_buf env);
 void grub_longjmp (grub_jmp_buf env, int val) __attribute__ ((noreturn));
 
 #endif /* ! GRUB_SETJMP_CPU_HEADER */
